pkg/porter/internal_plugins.go

```diff
@@ -0,0 +1,147 @@
+package porter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"get.porter.sh/porter/pkg/config"
+	"get.porter.sh/porter/pkg/plugins"
+	"get.porter.sh/porter/pkg/portercontext"
+	secretsplugins "get.porter.sh/porter/pkg/secrets/plugins"
+	"get.porter.sh/porter/pkg/secrets/plugins/filesystem"
+	"get.porter.sh/porter/pkg/secrets/plugins/host"
+	storageplugins "get.porter.sh/porter/pkg/storage/plugins"
+	"get.porter.sh/porter/pkg/storage/plugins/mongodb"
+	"get.porter.sh/porter/pkg/storage/plugins/mongodb_docker"
+	"get.porter.sh/porter/pkg/tracing"
+	"github.com/hashicorp/go-plugin"
+	"github.com/pkg/errors"
+)
+
+type RunInternalPluginOpts struct {
+	Key string
+}
+
+func (o *RunInternalPluginOpts) ApplyArgs(args []string) error {
+	if len(args) == 0 {
+		return errors.New("The positional argument KEY was not specified")
+	}
+	if len(args) > 1 {
+		return errors.New("Multiple positional arguments were specified but only one, KEY is expected")
+	}
+
+	o.Key = args[0]
+	return nil
+}
+
+func (o *RunInternalPluginOpts) Validate() error {
+	if o.Key == "" {
+		return fmt.Errorf("no plugin key was specified")
```

Comment:
We have an open issue ( #2009 ) to move off of the pkg/errors package and move to native errors. As I am working on the codebase, as I spot errors that don't rely on pkg/errors.Cause, I am switching them to native errors so that we can move to errors.Is eventually throughout.

We'll still need a sweep eventually of the entire codebase, but in the meantime I don't want to add to the tech debt.